### ISSUE TITLE
ghc: depends on libncursesw.so.5

### ISSUE THREAD

**Issue by jwhite007** [Linuxbrew/linuxbrew#882]
```
Last 15 lines from /home/whitj15/.cache/Homebrew/Logs/ghc/05.configure:
2016-03-03 21:30:44 +0000

./configure
--prefix=/home/whitj15/tmp/ghc20160303-14946-1q1emq2/ghc-7.10.3/binary
--with-gmp-includes=/toolbox/NGS/.linuxbrew_amd/Cellar/ghc/7.10.3b/libexec/integer-gmp/include
--with-gmp-libraries=/toolbox/NGS/.linuxbrew_amd/Cellar/ghc/7.10.3b/libexec/integer-gmp/lib
--with-ld=ld
--with-gcc=/toolbox/NGS/.linuxbrew_amd/bin/gcc-5

configure: WARNING: unrecognized options: --with-ld
checking for path to top of build tree... /home/whitj15/tmp/ghc--binary20160303-14946-1wk487s/ghc-7.6.3
ghc/stage2/build/tmp/ghc-stage2: error while loading shared libraries: libncursesw.so.5: cannot open shared object file: No such file or directory
Can't work out build platform
```
Logs are here... https://gist.github.com/jwhite007/d42b1a46ec51fefdfe1a

**Comment by sjackman** [comment]
`error while loading shared libraries: libncursesw.so.5`

Does `brew install ncurses` resolve this issue?

**Comment by jwhite007** [comment]
I have ncurses installed; however, the version is 6 and not 5.

**Comment by jwhite007** [comment]
```
brew upgrade r
```
... is also looking for libncursesw.so.5.

**Comment by sjackman** [comment]
Hmm. This is a challenge. `ghc` uses some binaries to bootstrap, and it looks as though those binaries depend on `libncursesw.so.5`. As a workaround, can you tell me whether the following symlink gets it working?
```sh
ln -s libncursesw.so.6 `brew --prefix`/lib/libncursesw.so.5
```

**Comment by sjackman** [comment]
Please open a separate ticket for the R issue and report `brew gist-logs r`.

**Comment by jwhite007** [comment]
The symlink trick works... no errors issued upon "brew install ghc." Are there any deleterious effects of doing this?

**Comment by sjackman** [comment]
It should be just fine. A reasonable fix would be in the `ghc` formula create the `libncursesw.so.5` symlink in lib in `install` and then remove it at the end after it's compiled. I think it's just a build time dependency, not a run time dep. I'll have to check that. Please do submit a pull request to do that if you like.

**Comment by sjackman** [comment]
@jwhite007 Are you able/interested in submitting a pull request to fix this issue? Add to `def install` something like the following (untested)
```ruby
ln_s Formula["ncurses"].lib/"libncursesw.so.6", lib/"libncursesw.so.5" if OS.linux?
```
and then at the end of `def install`
```ruby
rm lib/"libncursesw.so.5" if OS.linux?
```

**Comment by sjackman** [comment]
Also please add a comment to the code that explains (in one line) why this hack is necessary.

**Comment by jwhite007** [comment]
Yes, I'll work on this. Suffering from the flu at the moment. I'll try to get to it tomorrow.

**Comment by sjackman** [comment]
Get better first! Good luck with it.

**Comment by DoomHammer** [comment]
After reading with understanding I have found that similar problem is visible in `gnupg2`. Shall I administer the same cure?

**Comment by sjackman** [comment]
This particular fix is only necessary in the formula of `ghc` because it uses precompiled executables to bootstrap which depend on `libncursesw.so.5`. For `gnupg2`, `brew reinstall readline gnupg2` should fix it. You'll need to reinstall anything that depends on `ncurses`. See `brew uses --installed ncurses`

**Comment by DoomHammer** [comment]
Seems like doing the trick. Thanks!

**Comment by zbeekman** [comment]
Any hope of a bottled `ghc` ? This will circumvent the issue, unless installed from source

**Comment by sjackman** [comment]
`ghc` is not labelled as being `cellar :any`, meaning that it will only work if it's installed in a particular directory that it was built for, such as `/home/linuxbrew/.linuxbrew`. So, no, I guess not. The logic that tests whether a bottle is relocatable or not is automated and sometimes wrong. So, I suggest building a bottle in one `HOMEBREW_PREFIX`, installing it in a different `HOMEBREW_PREFIX` and reporting back whether that bottle works for you.

**Comment by kimleeng** [comment]
Just wanted to add the comment that preforming 
`brew install r --without-x11`
Also lacks libncursesw.so.5 and symlinking it to libncursesw.so.6  did the trick for me

**Comment by sjackman** [comment]
Thanks for the report, @kimleeng. You can if you like open a new issue over at https://github.com/Homebrew/homebrew-science/issues
Be sure to include a link to this issue if you do.

**Comment by jwhite007** [comment]
I'm still planning on contributing to this as you, Shaun, suggested above, and I'd like to make the proper edits to the rb; however, my company is in the process of moving right now, and it's been just shy of a shit storm. I've got several impending project deadlines, but after the dust settles, I'll start contributing more to linuxbrew.

**Comment by sjackman** [comment]
I believe this particular bug is fixed by 5503308, but future contributions are more than welcome, James! Good luck with the move.
